types/react-query/index.d.ts

```diff
@@ -289,6 +289,7 @@ export type PaginatedQueryResult<TResult> =
 export interface InfiniteQueryResult<TResult, TMoreVariable> extends QueryResultBase<TResult[]> {
     data: TResult[];
     isFetchingMore: boolean;
+    canFetchMore?: boolean;
```

Comment:
I think it should not be optional, it is always provided.

Reply:
However, after looking to source, it seems it is always provided but could be `undefined`.
```suggestion
    canFetchMore: boolean | undefined;
```